Sources/App/Middleware/PhotoURLMiddleware.swift

```diff
@@ -0,0 +1,68 @@
+import Vapor
+import HTTP
+
+final class PhotoURLMiddleware: Middleware {
+
+  let domain: String
+
+  init(_ domain: String) {
+    self.domain = domain
+  }
+
+  func respond(to request: Request, chainingTo next: Responder) throws -> Response {
+    let response = try next.respond(to: request)
+
+    if let array = response.json?.array {
+      // this is Event
+      
+      let arrayWithPhotoURL = try array.map { (element) -> StructuredData in
+        var result = element
+
+        if let photoWithPath = element[User.Keys.photoURL]?.string {
+          let photoURL = "\(domain)/\(photoWithPath)"
+          try result.set(User.Keys.photoURL, photoURL)
+        }
+
+        if let speakersPhoto  = element[Event.Keys.speakersPhotos]?.array {
+
+          let speakersPhotoURL = speakersPhoto.map { (speackerPhoto) -> String? in
```

Comment:
`map { speakerPhoto in` and you have a typo in `speackerPhoto`

Reply:
yes, thank you!